var/spack/repos/builtin/packages/mlperf-opencatalyst/package.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+from spack.package import *
+
+
+class MlperfOpencatalyst(PythonPackage, CudaPackage):
+    """Reference implementation for the MLPerf HPC OpenCatalyst DimeNet++ benchmark"""
+
+    homepage = "https://opencatalystproject.org/"
+    git = "https://github.com/mlcommons/hpc.git"
+
+    version("main", branch="main")
+
+    tags = ["proxy-app"]
+
+    depends_on("python@3.8:", type=("build", "run"))
+    depends_on("py-ase@3.21:3.21.9999", type=("build", "run"))
+    depends_on("py-matplotlib@3.3:3.3.9999", type=("build", "run"))
+    depends_on("py-pre-commit@2.10:2.10.9999", type=("build", "run"))
+    depends_on("py-pymatgen@2020.12.31", type=("build", "run"))
+    depends_on("py-torch@1.8.1", type=("build", "run"))
+    depends_on("py-pyyaml@5.4:5.4.9999", type=("build", "run"))
+    depends_on("py-tensorboard@2.4:2.4.9999", type=("build", "run"))
+    depends_on("py-tqdm@4.58:4.58.9999", type=("build", "run"))
+    depends_on("py-sphinx", type=("build", "run"))
+    depends_on("py-nbsphinx", type=("build", "run"))
+    depends_on("pandoc", type=("build", "run"))
+    depends_on("py-black", type=("build", "run"))
+    depends_on("py-torch+cuda", when="+cuda", type=("build", "run"))
+    depends_on("py-torch~cuda", when="~cuda", type=("build", "run"))
+    depends_on("py-torch-scatter+cuda", when="+cuda", type=("build", "run"))
+    depends_on("py-torch-scatter~cuda", when="~cuda", type=("build", "run"))
+    depends_on("py-torch-spline-conv+cuda", when="+cuda", type=("build", "run"))
+    depends_on("py-torch-spline-conv~cuda", when="~cuda", type=("build", "run"))
```

Comment:
I would remove `CudaPackage` since this package doesn't change how it's built if `+cuda` is requested. Dependencies should be controlled separately, either through `packages.yaml`:
```yaml
packages:
  all:
    variants: +cuda
```
or on the command line:
```console
$ spack install py-ocp-models ^py-torch+cuda
```

Reply:
That has bitten me countless times but following conventions: Done